src/Orleans.Runtime/Hosting/EndpointOptions.cs

```diff
@@ -8,25 +10,97 @@ namespace Orleans.Hosting
     public class EndpointOptions
     { 
         /// <summary>
-        /// The external IP address or host name used for clustering.
-        /// </summary>
-        public string HostNameOrIPAddress { get; set; }
-
-        /// <summary>
-        /// The IP address used for clustering. Will be inferred from <see cref="HostNameOrIPAddress"/> if this is not directly specified.
+        /// The IP address used for clustering.
         /// </summary>
         public IPAddress IPAddress { get; set; }
 
         /// <summary>
         /// The port this silo uses for silo-to-silo communication.
         /// </summary>
-        public int Port { get; set; }
+        public int SiloPort { get; set; }
 
         /// <summary>
         /// The port this silo uses for silo-to-client (gateway) communication. Specify 0 to disable gateway functionality.
         /// </summary>
         public int ProxyPort { get; set; }
```

Comment:
`ProxyPort` is a confusing name that's been used for historical reason that don't matter anymore. I think `GatewayPort` is the proper name here and everywhere else "proxy" is used.